File: stack/stackInJavaScript/stack.js
let MAX = 100;
let top = -1;
let arr = Array(MAX).fill(0);

console.log(arr);

function isEmpty() {
  return top < 0;
}

function push(element) {
  if (top >= MAX - 1) {
    return console.log("Stack is full");
  } else {
    top++;
    arr[top] = element;
    return console.log(element + " is popped into stack");
  }
}

function pop() {
  if (top >= MAX - 1) {
    return console.log("Stack is empty");
  } else {
    let x = arr[top];
    top--;
    return x;
  }
}

function print() {
  for (var i = top; i > -1; i--) {
    console.log(" " + arr[i]);
  }
}

push(100);
push(200);
push(300);
push(400);

pop();

print();

File: stack/stackInC/stack.c
#include<stdio.h>
#include <stdlib.h>


#define MAX 10


struct stack {
  int items[MAX];
  int top;
};

//create empty stack
void  crateEmptyStack(struct stack *s){
    s->top = -1;
}

// chack the stack is full
int isFull(struct stack *s){
    if (s->top == MAX-1)
      return 1;
    else
      return 0;
}

// chack the stack is empty
int isEmpty(struct stack *s){
    if(s->top == -1)
    return 1;
    else
    return 0;
}

//add element into the stack
void push(struct stack *s , int items){
    if(isFull(s)){
        printf("the stack is full\n");
    }
    else{
       s->top++;
       

    }
}
//remove elememt into the stack

//print the stack

int main(){


    return 0;
}